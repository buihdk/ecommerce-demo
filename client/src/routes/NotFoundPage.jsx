import React, { memo } from 'react';
import { useHistory } from 'react-router-dom';
import { makeStyles, Box, Typography, Button } from '@material-ui/core';

const useStyles = makeStyles({
  container: {
    height: '100vh',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  emoji: {
    fontSize: 40,
    letterSpacing: 10,
  },
});

const NotFoundPage = () => {
  const styles = useStyles();
  const history = useHistory();
  const handleClick = () => history.push('/');

  return (
    <Box className={styles.container}>
      <Typography variant="h1">Error 404</Typography>
      <Typography variant="h6" gutterBottom>
        Page not found.
      </Typography>
      <span className={styles.emoji} role="img" aria-label="sad-faces">
        ðŸ˜µðŸ˜–ðŸ˜«
      </span>
      <Button onClick={handleClick} variant="outlined">
        Go back to homepage
      </Button>
    </Box>
  );
};

export default memo(NotFoundPage);
