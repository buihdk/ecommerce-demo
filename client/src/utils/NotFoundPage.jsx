import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import { makeStyles, Box, Typography, Button } from '@material-ui/core';

const useStyles = makeStyles({
  container: {
    height: '100vh',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  emoji: {
    fontSize: 40,
    letterSpacing: 10,
  },
});

const NotFoundPage = ({ errorCode }) => {
  const styles = useStyles();
  const history = useHistory();
  const handleClick = () => history.push('/');

  let response = '';
  switch (errorCode) {
    case 404:
      response = 'Page not found.';
      break;
    case 400:
      response = 'Item does not exist.';
      break;
    default:
      response = 'Bad request.';
  }

  return (
    <Box className={styles.container}>
      <Typography variant="h1">{`Error ${errorCode}`}</Typography>
      <Typography variant="h6" gutterBottom>
        {response}
      </Typography>
      <span className={styles.emoji} role="img" aria-label="sad-faces">
        ðŸ˜µðŸ˜–ðŸ˜«
      </span>
      <Button onClick={handleClick} variant="outlined">
        Back to homepage
      </Button>
    </Box>
  );
};

NotFoundPage.propTypes = {
  errorCode: PropTypes.number,
};

NotFoundPage.defaultProps = {
  errorCode: 0,
};

export default memo(NotFoundPage);
