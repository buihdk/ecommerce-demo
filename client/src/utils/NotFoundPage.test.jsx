import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import NotFoundPage from './NotFoundPage';

const mockHistory = { push: jest.fn() };
jest.mock('react-router-dom', () => ({ useHistory: () => mockHistory }));

describe('NotFoundPage', () => {
  test('renders 404 without crashing', () => {
    const wrapper = shallow(<NotFoundPage errorCode={404} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('renders 400 without crashing', () => {
    const wrapper = shallow(<NotFoundPage errorCode={400} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('renders bad request without crashing', () => {
    const wrapper = shallow(<NotFoundPage />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('goes to homepage when button is clicked ', () => {
    const wrapper = shallow(<NotFoundPage />);
    wrapper.find('WithStyles(ForwardRef(Button))').simulate('click');

    expect(mockHistory.push).toBeCalledWith('/');
  });
});
