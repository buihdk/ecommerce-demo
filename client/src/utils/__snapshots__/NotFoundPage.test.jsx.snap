// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage renders 400 without crashing 1`] = `
<Styled(MuiBox)
  className="makeStyles-container-1"
>
  <WithStyles(ForwardRef(Typography))
    variant="h1"
  >
    Error 400
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    gutterBottom={true}
    variant="h6"
  >
    Item does not exist.
  </WithStyles(ForwardRef(Typography))>
  <span
    aria-label="sad-faces"
    className="makeStyles-emoji-2"
    role="img"
  >
    ğŸ˜µğŸ˜–ğŸ˜«
  </span>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
    variant="outlined"
  >
    Back to homepage
  </WithStyles(ForwardRef(Button))>
</Styled(MuiBox)>
`;

exports[`NotFoundPage renders 404 without crashing 1`] = `
<Styled(MuiBox)
  className="makeStyles-container-1"
>
  <WithStyles(ForwardRef(Typography))
    variant="h1"
  >
    Error 404
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    gutterBottom={true}
    variant="h6"
  >
    Page not found.
  </WithStyles(ForwardRef(Typography))>
  <span
    aria-label="sad-faces"
    className="makeStyles-emoji-2"
    role="img"
  >
    ğŸ˜µğŸ˜–ğŸ˜«
  </span>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
    variant="outlined"
  >
    Back to homepage
  </WithStyles(ForwardRef(Button))>
</Styled(MuiBox)>
`;

exports[`NotFoundPage renders bad request without crashing 1`] = `
<Styled(MuiBox)
  className="makeStyles-container-1"
>
  <WithStyles(ForwardRef(Typography))
    variant="h1"
  >
    Error undefined
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    gutterBottom={true}
    variant="h6"
  >
    Bad request.
  </WithStyles(ForwardRef(Typography))>
  <span
    aria-label="sad-faces"
    className="makeStyles-emoji-2"
    role="img"
  >
    ğŸ˜µğŸ˜–ğŸ˜«
  </span>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
    variant="outlined"
  >
    Back to homepage
  </WithStyles(ForwardRef(Button))>
</Styled(MuiBox)>
`;
